app/uk/gov/hmrc/hec/models/HECTaxCheck.scala

```diff
@@ -16,17 +16,20 @@
 
 package uk.gov.hmrc.hec.models
 
+import cats.implicits.catsSyntaxOptionId
 import play.api.libs.json.{Json, OFormat}
 
 import java.time.{LocalDate, ZonedDateTime}
+import java.util.UUID
 
 final case class HECTaxCheck(
   taxCheckData: HECTaxCheckData,
   taxCheckCode: HECTaxCheckCode,
   expiresAfter: LocalDate,
   createDate: ZonedDateTime,
   isExtracted: Boolean,
-  correctiveAction: Option[CorrectiveAction]
+  correctiveAction: Option[CorrectiveAction],
+  fileCorrelationId: Option[UUID] = UUID.randomUUID().some
```

Comment:
it'd be safer to not use default values here. When a new tax check gets written to the database, it makes more sense for the ID to be `None` since it's not part of any files

Reply:
I was of the view that all records will have some id, and when SDES notifies about them, their value will be changed to None. If initially all records have None, then does it change to UUID, when they are being written to file?
so will the process be like this: 1) read from db where isExtracted is false,2) write in file, with random UUID, update that record in db with same UUID, 3) send the file 4) sdes notifies them , then clean the UUID from db?
in this case there will be lot of DB hit

Reply:
or is it like 
: 1) read from db where isExtracted is false,2)add filecorrelation id for those records3) write records in file, 3) send the file 4) sdes notifies them , then clean the UUID from db?

Reply:
yes, it's like the latter thing you mentioned. All the tax checks in the same file need to have the same correlation id so it needs to be populated at the same time for all the tax checks in the scheduled job

Reply:
cool, got it now 👍

Reply:
updated [here](https://github.com/hmrc/hec/commit/a5e9a22ee554d55d6304279a09dd49216c727b7a)